src/features/session_trace/aggregate/index.js

```diff
@@ -44,65 +48,131 @@ export class Aggregate extends AggregateBase {
     this.harvestTimeSeconds = getConfigurationValue(agentIdentifier, 'session_trace.harvestTimeSeconds') || 10
     this.maxNodesPerHarvest = getConfigurationValue(agentIdentifier, 'session_trace.maxNodesPerHarvest') || 1000
     this.laststart = 0
+    this.isStandalone = false
+    const operationalGate = new HandlerCache() // acts as a controller-intermediary that can enable or disable this feature's collection dynamically
+    const sessionEntity = this.agentRuntime.session
+
+    /* --- The following section deals with user sessions concept & contains non-trivial control flow. --- */
+    const controlTraceOp = (traceMode) => {
+      switch (traceMode) {
+        case MODE.ERROR:
+        case MODE.FULL:
+        case true:
+          this.startTracing(operationalGate)
+          break
+        case MODE.OFF:
+        case false:
+        default: // this feature becomes "off" (does nothing & nothing is sent)
+          operationalGate.decide(false)
+          break
+      }
+    }
+    if (!sessionEntity) {
+      // Since session manager isn't around, do the old Trace behavior of waiting for RUM response to decide feature activation.
+      this.isStandalone = true
+      registerHandler('rumresp-stn', (on) => controlTraceOp(on), this.featureName, this.ee)
```

Comment:
Should probably just align on one way for waiting for flags.  Doesn't have to be done in this PR, but just below this you are using `waitForFlags()`.  For consistency would make sense to use that everywhere

Reply:
https://issues.newrelic.com/browse/NEWRELIC-9281

Reply:
tbh I used waitForFlags bc I needed _both_ flags in that case